test/micro/org/openjdk/bench/vm/compiler/CodeCacheStress.java

```diff
@@ -0,0 +1,445 @@
+                + "       } else {"
+                + "         return (Integer) m.get(k)+ instC;"
+                + "       }"
+                + "   }"
+                + " "
+                + "   public Integer get8( Map m, String k, Integer depth) { "
+                + "       if (depth > 0) {"
+                + "         return (Integer) m.get(k) + get9(m, k, --depth);"
+                + "       } else {"
+                + "         return (Integer) m.get(k)+ instD;"
+                + "       }"
+                + "   }"
+                + " "
+                + "   public Integer get9( Map m, String k, Integer depth) { "
+                + "       if (depth > 0) {"
+                + "         return (Integer) m.get(k) + get10(m, k, --depth);"
+                + "       } else {"
+                + "         return (Integer) m.get(k)+ instA;"
+                + "       }"
+                + "   }"
+                + " "
+                + "   public Integer get10( Map m, String k, Integer depth) { "
+                + "       if (depth > 0) {"
+                + "         return (Integer) m.get(k) + get11(m, k, --depth);"
+                + "       } else {"
+                + "         return (Integer) m.get(k)+ instB;"
+                + "       }"
+                + "   }"
+                + " "
+                + "   public Integer get11( Map m, String k, Integer depth) { "
+                + "       if (depth > 0) {"
+                + "         return (Integer) m.get(k) + get12(m, k, --depth);"
+                + "       } else {"
+                + "         return (Integer) m.get(k)+ instC;"
+                + "       }"
+                + "   }"
+                + " "
+                + "   public Integer get12( Map m, String k, Integer depth) { "
+                + "       if (depth > 0) {"
+                + "         return (Integer) m.get(k) + get(m, k, --depth);"
+                + "       } else {"
+                + "         return (Integer) m.get(k)+ instD;"
+                + "       }"
+                + "   }"
+                + "}";
+    }
+
+
+    class BenchLoader extends ClassLoader {
+
+        BenchLoader() {
+            super();
+        }
+
+        BenchLoader(ClassLoader parent) {
+            super(parent);
+        }
+
+        @Override
+        protected Class<?> findClass(String name) throws ClassNotFoundException {
+            if (name.equals(classNames[index])) {
+                assert compiledClasses[index] != null;
+                return defineClass(name, compiledClasses[index],
+                        0,
+                        (compiledClasses[index]).length);
+            } else {
+                return super.findClass(name);
+            }
+        }
+    }
+
+    CodeCacheStress.BenchLoader loader1 = new CodeCacheStress.BenchLoader();
+
+    final String k = "key";
+    final Integer v = 1000;
+
+    final String methodNames[] = {
+            "get"
+    };
```

Comment:
Does it have to be an array? There is only a single element, should it be a single `static final` field?

Reply:
In earlier versions it called more than 1 method in the generated classes, and I would prefer to keep that capability. It could be static, yes.